src/earn/prepareTransactions.ts

```diff
@@ -166,52 +173,79 @@ export async function prepareWithdrawTransactions({
   shortcutId: Exclude<EarnActiveAction, 'exit'>
   useMax: boolean
 }) {
-  const { dataProps } = pool
+  const { appId, balance, dataProps, networkId, shortcutTriggerArgs } = pool
   const { transactions }: { transactions: RawShortcutTransaction[] } = await triggerShortcutRequest(
     hooksApiUrl,
     {
       address: walletAddress,
-      appId: pool.appId,
-      networkId: pool.networkId,
-      shortcutId,
+      appId,
+      networkId,
+      shortcutId: 'withdraw',
       tokens: [
         {
           tokenId: dataProps.withdrawTokenId,
-          amount,
+          amount: useMax ? balance : amount,
           useMax,
         },
       ],
-      ...pool.shortcutTriggerArgs?.[shortcutId],
+      ...shortcutTriggerArgs?.withdraw,
     }
   )
 
   return {
     prepareTransactionsResult: await prepareTransactions({
       feeCurrencies,
       baseTransactions: rawShortcutTransactionsToTransactionRequests(transactions),
-      isGasSubsidized: isGasSubsidizedForNetwork(token.networkId),
-      origin: `earn-${shortcutId}`,
+      spendToken: token,
+      spendTokenAmount: new BigNumber(amount).shiftedBy(token.decimals),
+      isGasSubsidized: isGasSubsidizedForNetwork(networkId),
+      origin: 'earn-withdraw',
     }),
-    swapTransaction: undefined,
   }
 }
 
-export function usePrepareTransactions(mode: EarnActiveAction) {
-  const prepareTransactions = useAsyncCallback(
-    mode === 'withdraw' ? prepareWithdrawTransactions : prepareDepositTransactions,
-    {
-      onError: (err) => {
-        const error = ensureError(err)
-        Logger.error(TAG, 'usePrepareWithdrawTransactions', error)
-      },
-    }
-  )
+export async function prepareClaimTransactions({
+  pool,
+  walletAddress,
+  feeCurrencies,
+  hooksApiUrl,
+  rewardsPositions,
+}: {
+  pool: EarnPosition
+  walletAddress: Address
+  feeCurrencies: TokenBalance[]
+  hooksApiUrl: string
+  rewardsPositions: Position[]
+}) {
+  const { appId, networkId } = pool
 
+  // Prepare the claim transactions if there are rewards positions
+  const claimTransactions = rewardsPositions
+    ? await Promise.all(
```

Comment:
this checks if `rewardsPositions` is falsy, but can it ever be falsy since its guarenteed to be an array?

Reply:
Fixed in e9f2c075215324c83c03db3c107e9b1a62df6523.